### PR TITLE
[Assessor] Add filters on assessor request and vote places for the manager area

### PR DESCRIPTION
**Pull request by JeremieSamson**
assessor_request_filters

assessor_vote_place_filters

### CODE REVIEW

File: templates/assessor_manager/_filters.html.twig [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        <div style="width: 220px;">
+            <br />
+            <button type="submit" class="btn btn--blue b__nudge--right">Filtrer</button>
+            {% if filters.hasData %}
+                <a href="{{ path(app.request.get('_route'), app.request.get('_route_params')|merge({'status': filters.status})) }}">Annuler</a>
+            {% endif %}
+        </div>
+    </div>
+</form>
```
Comment (Remg) [inline templates/assessor_manager/_filters.html.twig]: bonsoir

File: src/Assessor/Filter/AssessorFilters.php [hunk 2]
```diff
@@ -63,6 +83,56 @@ public function getStatus(): ?string
...
+        }
+
+        if (!\in_array($country, array_keys($this->getCountries()), true)) {
+            throw new AssessorException(sprintf('Invalid country filter value given ("%s").', $country));
+        }
+
+        $this->country = trim($country);
+    }
+
+    public function getCountries(): array
```
Comment (thomasbisignani) [inline src/Assessor/Filter/AssessorFilters.php]: This method is really needed ?

What do you think to call directly `UnitedNationsBundle::getCountries()` in the setter : 
```php
if (!\in_array($country, array_keys(UnitedNationsBundle::getCountries()), true)) {
```

File: src/Assessor/Filter/AssessorRequestFilters.php [hunk 3]
```diff
@@ -55,6 +78,58 @@ public function apply(QueryBuilder $qb, string $alias): void
...
+        }
+
+        if ($this->getVotePlace()) {
+            if ($this->isStatusUnprocessed()) {
+                $qb->innerJoin("$alias.votePlaceWishes", 'vp');
+            } else {
+                $qb->innerJoin("$alias.votePlace", 'vp');
+            }
+
+            if (preg_match('/([0-9]{5}|2[A-Z][0-9]{3})_[0-9]{4}/', $this->getVotePlace())) {
```
Comment (thomasbisignani) [inline src/Assessor/Filter/AssessorRequestFilters.php]: What do you think to use a const variable to put the `/([0-9]{5}|2[A-Z][0-9]{3})_[0-9]{4}/` regex in the `AssessorFilters` class ?

File: src/Assessor/Filter/VotePlaceFilters.php [hunk 4]
```diff
@@ -52,6 +52,41 @@ public function apply(QueryBuilder $qb, string $alias): void
...
+
+        if ($this->getCountry()) {
+            $qb
+                ->andWhere("$alias.country = :country")
+                ->setParameter('country', $this->getCountry())
+            ;
+        }
+
+        if ($this->getVotePlace()) {
+            if (preg_match('/([0-9]{5}|2[A-Z][0-9]{3})_[0-9]{4}/', $this->getVotePlace())) {
```
Comment (thomasbisignani) [inline src/Assessor/Filter/VotePlaceFilters.php]: Same here to use a const

File: src/Assessor/Filter/AssessorFilters.php [hunk 5]
```diff
@@ -63,6 +85,56 @@ public function getStatus(): ?string
...
+        }
+
+        if (!\in_array($country, array_keys($this->getCountries()), true)) {
+            throw new AssessorException(sprintf('Invalid country filter value given ("%s").', $country));
+        }
+
+        $this->country = $country;
+    }
+
+    public function getCountries()
```
Comment (Remg) [inline src/Assessor/Filter/AssessorFilters.php]: missing return type

File: templates/assessor_manager/_filters.html.twig [hunk 6]
```diff
@@ -0,0 +1,44 @@
+<form method="get">
```
[the remark this answers is not in the document]
Comment (Remg) [reply templates/assessor_manager/_filters.html.twig]: he isnt using a FormView to render the form

[the remark this answers is not in the document]
Comment (Remg) [reply templates/assessor_manager/_filters.html.twig]: he isnt using a FormView to render the form

**Pull request merged**